rest-server/engines/src/main/scala/com/actionml/engines/cb/CBAlgorithm.scala

```diff
@@ -300,10 +287,33 @@ class CBAlgorithm(resourceId: String, dataset: CBDataset)
     item
   }
 
+  override def destroy(): Unit = CBAlgorithm.destroy(actors, vw, engineId, Paths.get(modelPath), params)
+
 }
 
 object CBAlgorithm extends LazyLogging {
 
+  def destroy(actors: ActorSystem, vw: VWMulticlassLearner, engineId: String, modelPath: java.nio.file.Path, params: CBAlgoParams): Unit = synchronized {
+    try {
+      Await.result(
+        actors.terminate().flatMap { _ =>
+          logger.info(s"Closing vw $vw")
+          if (vw != null.asInstanceOf[VWMulticlassLearner]) Future(vw.close())
+          else Future.successful(())
+        }.map { _ =>
+          if (Files.exists(modelPath) && !Files.isDirectory(modelPath)) {
+            logger.info(s"Deleting file ${modelPath} for vw $vw")
+            while (!Files.deleteIfExists(modelPath)) {logger.info(s"trying to delete ${modelPath}")}
+          } else logger.info(s"$vw has no file to delete")
+        }, 2 minutes)
+      logger.info(s"Algorithm for $engineId (created with params $params) was destroyed")
+    } catch {
+      case e: IOException => logger.error(s"Can't destroy $this", e)
+      case e: TimeoutException => logger.error(s"Error unable to delete the VW model file for $engineId at $modelPath in the 3 second timeout.", e)
+      case e => logger.error(s"Can't destroy algorithm for $engineId (created with params $params) - unknown error", e)
```

Comment:
this is dangerous, it catches all errors, some should be fatal

Reply:
yeah, I'll remove it